File: BM.BLL/Bookmanager.cs
using BookManagement.Model;
using BookManagement.DAL;
using System.Data;

namespace BookManagement.BLL
{
    public class BookManager
    {
        
        public DataTable GetBookById(int bookId)
        {
            return bookService.GetBookById(bookId);
        }

        public DataTable GetBookByISBN(string isbn)
        {
            return bookService.GetBookByISBN(isbn);
        }
        private BookService bookService = new BookService();

        public DataTable GetAllBooks()
        {
            return bookService.GetAllBooks();
        }

        public DataTable GetAllCategories()
        {
            return bookService.GetAllCategoriesForComboBox();
            return bookService.GetAllCategories();
        }

        public bool AddBook(Book book)
        {
            if (string.IsNullOrEmpty(book.BookName))
                throw new System.Exception("书名不能为空！");
            if (book.CategoryId <= 0)
                throw new System.Exception("请选择图书类别！");
            if (book.TotalCount <= 0)
                throw new System.Exception("图书总数必须大于0！");

            return bookService.AddBook(book) > 0;
        }

        public bool UpdateBook(Book book)
        {
            if (string.IsNullOrEmpty(book.BookName))
                throw new System.Exception("书名不能为空！");
            if (book.CategoryId <= 0)
                throw new System.Exception("请选择图书类别！");

            return bookService.UpdateBook(book) > 0;
        }

        public bool DeleteBook(int bookId)
        {
            // 实际开发中，这里应检查是否有未归还的借阅记录
            return bookService.DeleteBook(bookId) > 0;
        }
    }
}

File: Python/bll/book_manager.py
from dal.book_service import BookService
from models.book import Book


class BookManager:
    def __init__(self):
        self.book_service = BookService()

    def get_book_by_id(self, book_id: int):
        return self.book_service.get_book_by_id(book_id)

    def get_book_by_isbn(self, isbn: str):
        return self.book_service.get_book_by_isbn(isbn)

    def get_all_books(self):
        return self.book_service.get_all_books()

    def get_all_categories(self):
        return self.book_service.get_all_categories()

    def add_book(self, book: Book) -> bool:
        if not book.book_name:
            raise Exception("书名不能为空！")
        if book.category_id <= 0:
            raise Exception("请选择图书类别！")
        if book.total_count <= 0:
            raise Exception("图书总数必须大于0！")
        return self.book_service.add_book(book) > 0

    def update_book(self, book: Book) -> bool:
        if not book.book_name:
            raise Exception("书名不能为空！")
        if book.category_id <= 0:
            raise Exception("请选择图书类别！")
        return self.book_service.update_book(book) > 0

    def delete_book(self, book_id: int) -> bool:
        return self.book_service.delete_book(book_id) > 0